napari/layers/surface/surface.py

```diff
@@ -423,6 +444,34 @@ def _extent_data(self) -> np.ndarray:
             extrema = np.vstack([mins, maxs])
         return extrema
 
+    @property
+    def features(self):
+        """Dataframe-like features table.
+
+        It is an implementation detail that this is a `pandas.DataFrame`. In the future,
+        we will target the currently-in-development Data API dataframe protocol [1].
+        This will enable us to use alternate libraries such as xarray or cuDF for
+        additional features without breaking existing usage of this.
+
+        If you need to specifically rely on the pandas API, please coerce this to a
+        `pandas.DataFrame` using `features_to_pandas_dataframe`.
+
+        References
+        ----------
+        .. [1]: https://data-apis.org/dataframe-protocol/latest/API.html
+        """
+        return self._feature_table.values
+
+    @features.setter
+    def features(
+        self,
+        features: Union[Dict[str, np.ndarray], pd.DataFrame],
+    ) -> None:
+        self._feature_table.set_values(features, num_data=len(self.data))
+        self.text.refresh(self.features)
+        self.events.properties()
+        self.events.features()
```

Comment:
You'll need to manually add a `features` event in `self.events.add` in `__init__`.